Author: sj
Date: Wed May 13 12:59:08 2009
New Revision: 271862

Log:
#i101563# fixed crash when loading pptx document

Modified:
   cws/impress171/sd/source/ui/unoidl/unopback.cxx

Modified: cws/impress171/sd/source/ui/unoidl/unopback.cxx
==============================================================================
--- sd/source/ui/unoidl/unopback.cxx	Wed May 13 12:39:35 2009	(r271861)
+++ sd/source/ui/unoidl/unopback.cxx	Wed May 13 12:59:08 2009	(r271862)
@@ -140,22 +140,56 @@
 				if( pAny )
 				{
 					OUString aPropertyName( OUString::createFromAscii(pMap->pName));
-					if ( pMap->nWID == XATTR_FILLBITMAP )
+					switch( pMap->nWID )
 					{
-						if ( ( ( pAny->getValueType() == ::getCppuType((const ::com::sun::star::uno::Reference< ::com::sun::star::awt::XBitmap >*)0) ) ||
-								( pAny->getValueType() == ::getCppuType((const ::com::sun::star::uno::Reference< ::com::sun::star::graphic::XGraphic >*)0) ) ) &&
-								( pMap->nMemberId == MID_BITMAP ) )
+						case XATTR_FILLFLOATTRANSPARENCE :
+						case XATTR_FILLGRADIENT :
 						{
-							setPropertyValue( aPropertyName, *pAny );
+							if ( ( pAny->getValueType() == ::getCppuType((const ::com::sun::star::awt::Gradient*)0) )
+								&& ( pMap->nMemberId == MID_FILLGRADIENT ) )
+							{
+								setPropertyValue( aPropertyName, *pAny );
+							}
+							else if ( ( pAny->getValueType() == ::getCppuType((const ::rtl::OUString*)0) ) &&
+										( pMap->nMemberId == MID_NAME ) )
+							{
+								setPropertyValue( aPropertyName, *pAny );
+							}
 						}
-						else if ( ( pAny->getValueType() == ::getCppuType((const ::rtl::OUString*)0) ) &&
-									( ( pMap->nMemberId == MID_NAME ) || ( pMap->nMemberId == MID_GRAFURL ) ) )
+						break;
+						case XATTR_FILLHATCH :
 						{
-							setPropertyValue( aPropertyName, *pAny );
+							if ( ( pAny->getValueType() == ::getCppuType((const ::com::sun::star::drawing::Hatch*)0) )
+								&& ( pMap->nMemberId == MID_FILLHATCH ) )
+							{
+								setPropertyValue( aPropertyName, *pAny );
+							}
+							else if ( ( pAny->getValueType() == ::getCppuType((const ::rtl::OUString*)0) ) &&
+										( pMap->nMemberId == MID_NAME ) )
+							{
+								setPropertyValue( aPropertyName, *pAny );
+							}
 						}
-					}
-					else
-						setPropertyValue( aPropertyName, *pAny );
+						break;
+						case XATTR_FILLBITMAP :
+						{
+							if ( ( ( pAny->getValueType() == ::getCppuType((const ::com::sun::star::uno::Reference< ::com::sun::star::awt::XBitmap >*)0) ) ||
+									( pAny->getValueType() == ::getCppuType((const ::com::sun::star::uno::Reference< ::com::sun::star::graphic::XGraphic >*)0) ) ) &&
+									( pMap->nMemberId == MID_BITMAP ) )
+							{
+								setPropertyValue( aPropertyName, *pAny );
+							}
+							else if ( ( pAny->getValueType() == ::getCppuType((const ::rtl::OUString*)0) ) &&
+										( ( pMap->nMemberId == MID_NAME ) || ( pMap->nMemberId == MID_GRAFURL ) ) )
+							{
+								setPropertyValue( aPropertyName, *pAny );
+							}
+						}
+						break;
+
+						default:
+							setPropertyValue( aPropertyName, *pAny );
+					}					
 				}
 				pMap++;
 			}

---------------------------------------------------------------------
To unsubscribe, e-mail: allsvn-unsubscribe@openoffice.org
For additional commands, e-mail: allsvn-help@openoffice.org

